### PR TITLE
Update Validations for Host Rental Application

### PR DESCRIPTION
**Pull request by dimak1**
*Issue:*

- bcgov/entity/issues/23955
- bcgov/entity/issues/23810
- bcgov/entity/issues/24231
- bcgov/entity/issues/24222
- bcgov/entity/issues/23769
- bcgov/entity/issues/23953

*Description of changes:*
- Refactor validations to use Nuxt built-in validation
- Update some rules
- Fix validation bugs
- Add unit tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-ui-dev--pr-306-v8lyzsa9.web.app

File: strr-web/stores/strr.ts [hunk 1]
```diff
@@ -12,18 +12,27 @@ import { HostContactTypeE } from '~/enums/host-contact-type-e'
...
+const httpRegex = /^(https?:\/\/)?([\w-]+(\.[\w-]+)+\.?(:\d+)?(\/.*)?)$/i // https is optional
 const html5EmailRegex = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)+$/
 const html5PhoneRegex = /^(\+\d{1,3}[-\s.]?)?(\(?\d{3}\)?[-\s.]?\d{3}[-\s.]?\d{4,6})$/
 const pidRegex = /^\d{3}(-)\d{3}(-)\d{3}$/
-const sinRegex = /^\d{3}( )\d{3}( )\d{3}$/
-const craBusinessNumberRegex = /^\d{9}$|^\d{9}[A-Z]{2}\d{4}$/
+
+// not allowed: all 0s, starting with 8, letters; allowed: 111 222 333 or 111222333
+const sinRegex = /^(?!00000000|000 000 000)(?!8)(?:\d{9}|\d{3} \d{3} \d{3})$/
+const craBusinessNumberRegex = /^\d{9}$/
```
Comment (dimak1) [inline strr-web/stores/strr.ts:22]: Updated business number validation as per ticket 24222

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-ui-dev--pr-306-v8lyzsa9.web.app

**Comment by dimak1** [comment]
Ready for review @shaangill025 @kris-daxiom @rstens

**Review by shaangill025 — APPROVED** [review]
LGTM

**Pull request merged**